File: src/com/manulaiko/psfucker/Fingerprinter.java
package com.manulaiko.psfucker;

/**
 * Base interface for all servers fingerprinters
 *
 * @author Manulaiko
 * @package com.manulaiko
 */
public abstract class Fingerprinter
{
    /**
     * Server fucker
     *
     * @var com.manulaiko.Fucker
     */
    public Fucker fucker = null;

    /**
     * Starts fingerprinting the server
     *
     * @return boolean True if server could be identified
     */
    public abstract boolean identify(String server);
}

File: src/main/kotlin/com/manulaiko/psfucker/Fingerprinter.kt
package com.manulaiko.psfucker

/**
 * Fingerprinter interface.
 * ========================
 *
 * Base interface for all servers fingerprinters.
 *
 * @author Manulaiko <manulaiko@gmail.com>
 */
interface Fingerprinter<T : Fucker> {
    /**
     * The fucker for this kind of server.
     */
    val fucker: T

    /**
     * Starts fingerprinting the server
     *
     * @return True if server could be identified
     */
    fun identify(server: String): Boolean
}
